Snowflake.Data/Core/RestRequester.cs

```diff
@@ -48,7 +48,19 @@ static internal RestRequester Instance
         public T Post<T>(IRestRequest request)
         {
             //Run synchronous in a new thread-pool task.
-            return Task.Run(async () => await PostAsync<T>(request, CancellationToken.None)).Result;
+            var task = Task.Run(async () => await PostAsync<T>(request, CancellationToken.None));
+            try
+            {
+                return task.Result;
+            }
+            catch (AggregateException ae)
+            {
+                foreach (var ex in ae.InnerExceptions)
+                {
+                    logger.Error("Rest request failure", ex);
+                }
+                throw ae.GetBaseException();
```

Comment:
I would have to write a test case, but I'm pretty sure `ae.GetBaseException()` is likely to just be the `ae` instance, in which case this doesn't give us any benefit over simply just using `throw` to rethrow the exception and it will clobber the stack trace on the way out...